src/database/services/PanelSetService.ts

```diff
@@ -0,0 +1,47 @@
+import { Sequelize } from "sequelize";
+import { PanelSet } from "../models/panelSet.model";
+import { User } from "../models/user.model";
+
+interface PanelSetConfig {
+    username: string
+}
+
+class PanelSetService {
+    /**
+     * Create a new panel set
+     * @param {} panelSet 
+     */
+    static async createPanelSet(panelSet: PanelSetConfig) {
+        // get id from username
+        const user = await User.findOne({ where: {username: panelSet.username } });
+        if (user == null) return undefined;
+
+        const { author_id } = await PanelSet.create({
+            author_id: user.id
+        });
+
+        return { author_id };
+    }
+    /**
+     * Gets a panel set based on the id
+     * @param {} id the id of the panel set
+     * @returns null if the panel set is not found/exists
+     */
+    static async getPanelSetByID(id: number) {
+        return await PanelSet.findByPk(id);
+    }
+
+    /**
+     * Get all of the panels a specific author created
+     * @param {} username author's username
+     * @returns an array of all the panels found
+     */
+    static async getAllPanelSetFromUser(username: string) {
+        return await User.findAll({
+            where: { username },
+            include: { model: PanelSet },
+        });
```

Comment:
I don't know if this is a good suggestion as things could change depending on the relationships between models, but could potentially map the data of the returned array to reduce unnecessary data like created_at from being returned.

Reply:
I assume so. This is the SQL equivalent ``SELECT panel_sets.id, panel_sets.created_at, panel_sets.updated_at FROM panel_sets INNER JOIN users ON panel_sets.author_id = users.id WHERE users.username = username;``, but I'm unsure on how to parse this into sequalize.

Reply:
I think you could save into an array then map it to filter the unneeded data after the fetching. I am also unsure of how to select specific attributes from `findAll` in sequelize. Either way, not a huge deal.